src/scintillaeditor.cpp

```diff
@@ -1208,3 +1208,13 @@ void ScintillaEditor::jumpToNextError()
 {
 	findMarker(1, 0, [this](int line){ return qsci->markerFindNext(line, 1 << errMarkerNumber); });
 }
+
+void ScintillaEditor::stringToEndOfTheLine(){
+    if(qsci->hasFocus()){
+        int line,index,line_len;
+        qsci->getCursorPosition(&line,&index);
+        line_len = qsci->lineLength(line);
```

Comment:
This calculation seems to be wrong in case of characters that are more than one byte, e.g. with cursor after the `>` and then `CTRL+K` gives:

![grafik](https://user-images.githubusercontent.com/1330241/75095633-3670b480-5597-11ea-817c-1f5c46532771.png)

![grafik](https://user-images.githubusercontent.com/1330241/75095640-45effd80-5597-11ea-850f-f53bb3a4e8df.png)

Reply:
Yeah, that lineLength method has got some problem while dealing with characters having size of more than one byte. But the problem is there is no other method available that give us the exact character count. I have also tried using positionFromLineIndex method which return the byte index in the whole text. But this will not help us because the setSelection method needs two pairs of line and relative position as input.

Reply:
Right, there's a hint in the docu of [`lineLength()`](https://www.riverbankcomputing.com/static/Docs/QScintilla/classQsciScintilla.html#ab5ecfa6bee9e78e5306986367e1194af): _In order to get the length in characters use text(line).length()._ 

I guess it should work factoring [this calculation](https://github.com/openscad/openscad/blob/5696d0703583d3d53b34683bbc59d3b1cb284a78/src/scintillaeditor.cpp#L1210) out into a separate function and use that. This makes sure to ignore the trailing newline if it exists.